torch/_utils.py

```diff
@@ -393,3 +393,59 @@ def reraise(self):
             # (https://bugs.python.org/issue2651), so we work around it.
             msg = KeyErrorMessage(msg)
         raise self.exc_type(msg)
+
+def _get_available_device_type():
```

Comment:
Looks like this can only return "cuda" or None? Does it mean I need to modify all functions here if I want to introduce a new device type? Or is that somehow dynamically loaded?

Reply:
Yes, you need to modify all the functions here for now. It is the simplest way to add a new device type manually without a device abstraction in python. 
The `torch.device` is a class instead of python module while there is a dedicate `torch.cuda` module for NV cards.

It might be better we can add the dynamically loading of devices with the abstraction of `torch.device`. Maybe like, 'torch.device.gpu.cuda', 'torch.device.fpga.xxx' or 'torch.device.xla'. 
Or do you have any suggestion that we should add the dynamically load somehow now?